src/docs/extensions.md

```diff
@@ -2,16 +2,27 @@
+- **Theia plugins**: Like VS Code extensions, additional access to some Theia specific APIs and the Frontend (frontend plugins), Theia specific parts are not compatible with VS Code.
+The following diagram shows the high level architecture for all three options. VS Code extensions and Theia plugins run in a dedicated process, can be installed at runtime and work against a defined API. Theia extensions are added during compile time and become a first class citizen within your Theia application. They can access the full API of Theia.
 
-Through a DI module, the extension can provide bindings from types to concrete
-implementations, i.e. provide services and contributions.
+<img src="/extensiontypes.png" alt="Theia Logo Blue" style="max-width: 525px">
+
+If you would like more guidance on which mechanism to use, please also refer to [this detailed comparions between VS Code extensions and Theia extensions](https://eclipsesource.com/blogs/2021/03/24/vs-code-extensions-vs-theia-extensions/).
+
+##Theia extensions
+A Theia extension is a module that resides inside a Theia application and communicates with other modules (Theia extensions) via dependency injection. The Theia project itself is composed of Theia extensions too. To create a Theia application, you can select a number of Theia extensions provided by the Theia project, add your own custom Theia extensions and then compile and run the result. Your custom Theia extension will have the same level of access as the core ones. This modularity allows you to extend, adapt or remove almost anything in Theia according to your requirements. Also specific use cases, such as complex views are easier to develop with Theia extensions compared to VS Code extensions.
+Technically, an extension is an npm package that exposes any number of DI modules (`ContainerModule`) that contribute to the creation of the DI container.
+An extension is consumed by adding a dependency to the npm-package in the `package.json` of the application. Extensions are meant to be installed at compile time
+See [this section](https://theia-ide.org/docs/authoring_extensions/) for more detail on how to author a Theia extension.
+##VS Code extensions
+VS Code extensions are the popular mechanism to extend VS Code with new language support and other features. VS Code extensions are simple to develop, they have access to a defined and restricted API and VS Code extensions can be installed at runtime. Eclipse Theia provides the same extension API as VS Code, so extensions are compatible. Therefore, to develop your own extension, please refer to the [VS Code extension documentation](https://code.visualstudio.com/api). Please also refer to this coverage report, highlighting which API of VS Code is covered by Theia.
+Please also note that you can use existing VS Code extensions in Theia, too. A good source for installing or downloading extensions is the [Open VSX registry](https://open-vsx.org/).
+##Theia plugins
```

Comment:
Please include a newline as well:

```suggestion

## Theia Plugins
```

Reply:
+1 to suggested capitalization of headings. I usually adhere to these rules:
https://www.webucator.com/article/how-to-capitalize-headings-and-titles/